.github/workflows/regression_tests.yml

```diff
@@ -5,17 +5,33 @@ on:
     branches: [ main ]
 
 jobs:
-  build_for_testing:
-    uses: ./.github/workflows/build_for_testing.yml
-    with:
-      developer-dir: /Applications/Xcode_14.2.0.app/Contents/Developer
-    secrets: inherit
+  execute_merge_tests:
+    runs-on: macos-latest
+    name: Execute Merge Tests
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_14.2.0.app/Contents/Developer
 
-  execute_regression_tests:
-    name: Execute regression tests
-    needs: build_for_testing
-    uses: ./.github/workflows/execute_test_plan.yml
-    with:
-      test-plan: MainTestPlan
-      developer-dir: /Applications/Xcode_14.2.0.app/Contents/Developer
-    secrets: inherit
\ No newline at end of file
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare environment
+        uses: ./.github/actions/prepare_environment
+        with:
+          core-token: ${{ secrets.IOS_CORE_TOKEN }}
+  
+      - name: Evaluate Tests to execute
+        id: evaluate-tests-to-execute
+        uses: ./.github/actions/evaluate_tests_to_execute
+
+      - name: Perform Tests and Publish Test reports
+        id: perform-tests
+        uses: ./.github/actions/perform_tests
+        with:
+          needs-shared-tests: ${{ true }}
```

Comment:
Why are all of these true? If that is the case I guess there is no need to execute `evaluate-tests-to-execute`, right?

Reply:
Correct! Didn't want to make another logic for regression tests only as it was easier to just copy paste the logic here as the `evaluate-tests-to-execute` takes < 1s in most pipelines

Reply:
Makes sense to re-use the action with all true 👍 what I mean is that in this case you can completely remove the evalutation since it is not used, can't you?